### PR TITLE
[PR] Create extract basic tests

### PR DESCRIPTION
**Pull request by rafsandwich**
# Description

Created a `test_extract.py` file with 8 basic, passing, tests.  

Note: This should be picked up again in the future - I know I'm pulled on other things atm but there is a refine ticket [#34] where I would like to see more tests for the api call, like a bad response, and that it's only called once. Also more edge cases for the other functions.

Closes: #30 

## Type of change

- [ ] New feature (change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

# Mentions
@AhmetCengiz03
@bxrkxnt
@rafsandwich
@JamezDon

### CODE REVIEW

File: pipeline/scrape_live_game.py
Comment (AhmetCengiz03) [inline pipeline/scrape_live_game.py:1]: Have you just changed the name of the file here, this looks like stuff we've already reviewes

File: pipeline/test_extract.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# pylint: skip-file
+"""Tests for the test_extract.py script."""
+
+import pytest
```
Comment (AhmetCengiz03) [inline pipeline/test_extract.py:4]:
```suggestion
from pytest import raises
```

I'll make the suggestions for the bits you need

File: pipeline/test_extract.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+def test_build_url_builds_correctly_with_id():
+    url = build_scrape_url(19194929, "MYTOKEN")
+
+    assert "fixtures/19194929?" in url
+    assert "api_token=MYTOKEN" in url
+
+
+def test_build_url_type_error_with_bad_type():
+
+    with pytest.raises(TypeError):
```
Comment (AhmetCengiz03) [inline pipeline/test_extract.py:61]:
```suggestion
    with raises(TypeError):
```

**Review by AhmetCengiz03 — APPROVED** [review]
I think this is great work Raf, could be worth writing a ticket to do a couple of more tests if we get time.

File: pipeline/test_extract.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    mock_conn = MagicMock()
+    mock_response = MagicMock()
+    mock_data = MagicMock()
+
+    mock_conn.getresponse.return_value = mock_response
+    mock_response.read.return_value = mock_data
+
+    mock_data.decode.return_value = '{"match_id": "101010", "fixture_name": "Scotland vs Hungary"}'
+    mock_response.status = 200
```
[the remark this answers is not in the document]
Comment (AhmetCengiz03) [reply pipeline/test_extract.py:75]: I wouldn't worry about it for now it works, we can defo refine later.
